File: src/day2/day2.cpp
#include <iostream>
#include <fstream>

using namespace std;

int solutions(string filename) {
	ifstream file;
	file.open(filename);

	string direction = "";
	int value = 0, horizontal = 0, depth = 0;

	// parse file and do operations
	if (file.is_open()) {
		while (file >> direction >> value) {
			if (direction == "forward") {
				horizontal += value;
			} else if (direction == "down") {
				depth += value;
			} else if (direction == "up") {
				depth -= value;
			}
		}
	}

	return horizontal * depth;
}

int main() {
	cout << solutions("input.txt") << endl;
}

File: src/day2/day2.clj
(ns AoC.day2
  (:require
    [clojure.java.io :as io]
    [clojure.string :as str]
    [clojure.set :as set]))

(defn parse-long [s]
  (Long/valueOf s))

(def data
  (with-open [rdr (io/reader (io/file "input.txt"))]
    (->> (line-seq rdr)
         ;; idk why there's a _ here
         (mapv #(let [[_ cmd arg] (re-matches #"([a-z]+) ([0-9]+)" %)]
                  [(keyword cmd) (parse-long arg)])))))

(defn part1 []
  (let [[x y] (reduce 
                (fn [[x y] [cmd arg]]
                  (case cmd
                    :forward [(+ x arg) y]
                    :down    [x (+ y arg)]
                    :up      [x (- y arg)]))
                [0 0] data)]
    (* x y)))

